src/AutoRest.CSharp/MgmtTest/AutoRest/MgmtTestTarget.cs

```diff
@@ -43,20 +41,20 @@ public static async Task ExecuteAsync(GeneratedCodeWorkspace project, InputNames
                 library = new MgmtTestOutputLibrary(inputNamespace);
             }
 
-            if (Configuration.MgmtTestConfiguration.Mock)
+            if (Configuration.MgmtTestConfiguration?.Mock ?? false)
             {
                 WriteMockTests(project, library);
             }
 
-            if (Configuration.MgmtTestConfiguration.Sample)
+            if (Configuration.MgmtTestConfiguration?.Sample ?? Configuration.GenerateSampleProject)
             {
                 WriteSamples(project, library);
             }
 
             if (_overriddenProjectFilenames.TryGetValue(project, out var overriddenFilenames))
                 throw new InvalidOperationException($"At least one file was overridden during the generation process. Filenames are: {string.Join(", ", overriddenFilenames)}");
 
-            if (Configuration.MgmtTestConfiguration.ClearOutputFolder)
+            if (Configuration.MgmtTestConfiguration?.ClearOutputFolder ?? true)
```

Comment:
why not default to false like L44?

Reply:
it might cause funny issues if it default to false, such as if we generate the MPG library, and add some `@clientName` to rename something, the types with old name will not be cleaned up, one might have issues.

Currently because we do not have a way to populate these useful configurations when the input is typespec, I have to choose a default value for them.

Reply:
Oh - the default value of this is `false` in `MgmtTestConfiguration`.
Let me change this to `false`. But I think it is problematic.